server/src/internalClusterTest/java/org/opensearch/wlm/WorkloadManagementStatsIT.java

```diff
@@ -0,0 +1,315 @@
+    }
+
+    public void testWlmStatsWithIdAndBreach() throws Exception {
+        QueryGroup queryGroup = new QueryGroup(
+            NAME1,
+            new MutableQueryGroupFragment(MutableQueryGroupFragment.ResiliencyMode.ENFORCED, Map.of(ResourceType.CPU, 0.5))
+        );
+        String queryGroupId = queryGroup.get_id();
+        String nodeId = client().admin().cluster().prepareState().clear().setNodes(true).get().getState().nodes().getLocalNodeId();
+        updateQueryGroupInClusterState(PUT, queryGroup);
+        WlmStatsResponse response = getWlmStatsResponse(new String[] { nodeId }, new String[] { DEFAULT_QUERY_GROUP }, true);
+        validateResponse(response, new String[] { nodeId }, new String[] { DEFAULT_QUERY_GROUP, queryGroupId });
+
+        WlmStatsResponse response2 = getWlmStatsResponse(null, new String[] { DEFAULT_QUERY_GROUP, queryGroupId }, true);
+        validateResponse(response2, null, new String[] { DEFAULT_QUERY_GROUP, queryGroupId });
+
+        WlmStatsResponse response3 = getWlmStatsResponse(null, new String[] { DEFAULT_QUERY_GROUP, queryGroupId }, false);
+        validateResponse(response3, new String[] { DEFAULT_QUERY_GROUP, queryGroupId }, null);
+
+        WlmStatsResponse response4 = getWlmStatsResponse(null, new String[] { queryGroupId }, false);
+        validateResponse(response4, new String[] { queryGroupId }, new String[] { DEFAULT_QUERY_GROUP });
+
+        updateQueryGroupInClusterState(DELETE, queryGroup);
+    }
+
+    public void updateQueryGroupInClusterState(String method, QueryGroup queryGroup) throws InterruptedException {
+        ExceptionCatchingListener listener = new ExceptionCatchingListener();
+        client().execute(TestClusterUpdateTransportAction.ACTION, new TestClusterUpdateRequest(queryGroup, method), listener);
+        assertTrue(listener.getLatch().await(TIMEOUT.getSeconds(), TimeUnit.SECONDS));
+        assertEquals(0, listener.getLatch().getCount());
+    }
+
+    public WlmStatsResponse getWlmStatsResponse(String[] nodesId, String[] queryGroupIds, Boolean breach) throws ExecutionException,
+        InterruptedException {
+        WlmStatsRequest wlmStatsRequest = new WlmStatsRequest(nodesId, new HashSet<>(Arrays.asList(queryGroupIds)), breach);
+        return client().execute(WlmStatsAction.INSTANCE, wlmStatsRequest).get();
+    }
+
+    public void validateResponse(WlmStatsResponse response, String[] validIds, String[] invalidIds) {
+        assertNotNull(response.toString());
+        String res = response.toString();
+        if (validIds != null) {
+            for (String validId : validIds) {
+                assertTrue(res.contains(validId));
+            }
+        }
+        if (invalidIds != null) {
+            for (String invalidId : invalidIds) {
+                assertFalse(res.contains(invalidId));
+            }
+        }
+    }
+
+    public static class TestClusterUpdateRequest extends ActionRequest {
```

Comment:
Lets make this class from https://github.com/opensearch-project/OpenSearch/pull/16359 or vice-versa based on the merge order to avoid code dulplicity

Reply:
Yes there are many duplicate classes in this and #16359. I'll remove the duplication once any of them is merged